pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -155,9 +157,12 @@ public QueryPlannerResult planQuery(String sqlQuery, SqlNodeAndOptions sqlNodeAn
     try (PlannerContext plannerContext = new PlannerContext(_config, _catalogReader, _typeFactory, _hepProgram)) {
       plannerContext.setOptions(sqlNodeAndOptions.getOptions());
       RelRoot relRoot = compileQuery(sqlNodeAndOptions.getSqlNode(), plannerContext);
-      Set<String> tableNames = getTableNamesFromRelRoot(relRoot.rel);
-      return new QueryPlannerResult(toDispatchablePlan(relRoot, plannerContext, requestId, tableNames), null,
-          tableNames);
+      SubPlan subPlan = toSubPlan(relRoot);
+      // TODO: current code only assume one SubPlan per query, but we should support multiple SubPlans per query.
+      // Each SubPlan should be able to run independently from Broker then set the results into the dependent
+      // SubPlan for further processing.
+      DispatchableSubPlan dispatchableSubPlan = toDispatchableSubPlan(subPlan, plannerContext, requestId);
+      return new QueryPlannerResult(dispatchableSubPlan, null, dispatchableSubPlan.getTableNames());
```

Comment:
i would all the first argument `rootSubPlan` or `entrypointSubPlan` and change the QueryPlannerResult member variable accordingly

Reply:
QueryPlannerResult is just the DispatchableSubPlan for one SubPlan (subplanRoot).